src/components/WelcomeText.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import {Text, View} from 'react-native';
+import styles from '../styles/styles';
+import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
+
+const propTypes = {
+    ...windowDimensionsPropTypes,
+};
+
+const WelcomeText = props => (
+    <View style={props.isSmallScreenWidth ? [] : [styles.mb6, styles.mt6]}>
```

Comment:
What if instead of using `withWindowDimensions` we just define the margin stuff in the parent component using it so that things are more flexible and then just end up with two different text modes that we can pass via a prop. We'd have to use a fragment, but it would make this component easier to use.

e.g. if you are in a narrow view then you can show the varied text, but this component doesn't _need_ to know that its layout has a relationship to screen width. Just that it has two different text modes. Here's how it might look...

```jsx
<View style={[styles.mb6, styles.mt6]}>
    <WelcomeText large />
</View>
```

and 

```jsx
<View>
    <WelcomeText />
</View>
```